In [1]:
# TUGAS PRAKTIKUM - DATA UNDERSTANDING
# 42324056 - Christina Adelia Sitinjak

## Tugas 1
Analisa korelasi antara `temperature_fahrenheit`, `pressure_mb`, dan `air_quality_Sulphur_dioxide` (dataset `GlobalWeatherRepository.csv`)

In [ ]:
import pandas as pd
path = "GlobalWeatherRepository.csv"
df = pd.read_csv(path)
df

,country,location_name,latitude,longitude,timezone,last_updated_epoch,last_updated,temperature_celsius,temperature_fahrenheit,condition_text,...,air_quality_PM2.5,air_quality_PM10,air_quality_us-epa-index,air_quality_gb-defra-index,sunrise,sunset,moonrise,moonset,moon_phase,moon_illumination
0,Afghanistan,Kabul,34.5200,69.1800,Asia/Kabul,1715849100,5/16/2024 13:15,26.6,79.8,Partly Cloudy,...,8.400,26.600,1,1,4:50 AM,6:50 PM,12:12 PM,1:11 AM,Waxing Gibbous,55
1,Albania,Tirana,41.3300,19.8200,Europe/Tirane,1715849100,5/16/2024 10:45,19.0,66.2,Partly cloudy,...,1.100,2.000,1,1,5:21 AM,7:54 PM,12:58 PM,2:14 AM,Waxing Gibbous,55
2,Algeria,Algiers,36.7600,3.0500,Africa/Algiers,1715849100,5/16/2024 9:45,23.0,73.4,Sunny,...,10.400,18.400,1,1,5:40 AM,7:50 PM,1:15 PM,2:14 AM,Waxing Gibbous,55
3,Andorra,Andorra La Vella,42.5000,1.5200,Europe/Andorra,1715849100,5/16/2024 10:45,6.3,43.3,Light drizzle,...,0.700,0.900,1,1,6:31 AM,9:11 PM,2:12 PM,3:31 AM,Waxing Gibbous,55
4,Angola,Luanda,-8.8400,13.2300,Africa/Luanda,1715849100,5/16/2024 9:45,26.0,78.8,Partly cloudy,...,183.400,262.300,5,10,6:12 AM,5:55 PM,1:17 PM,12:38 AM,Waxing Gibbous,55
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
39589,Venezuela,Caracas,10.5000,-66.9167,America/Caracas,1733482800,12/6/2024 7:00,16.2,61.2,Sunny,...,14.430,19.240,1,2,6:33 AM,6:05 PM,11:11 AM,11:08 PM,Waxing Crescent,24
39590,Vietnam,Hanoi,21.0333,105.8500,Asia/Bangkok,1733481000,12/6/2024 17:30,21.2,70.2,Partly cloudy,...,99.160,100.825,4,10,6:20 AM,5:15 PM,10:35 AM,9:55 PM,Waxing Crescent,20
39591,Yemen,Sanaa,15.3547,44.2067,Asia/Aden,1733481900,12/6/2024 13:45,22.2,71.9,Sunny,...,10.360,31.080,1,1,6:16 AM,5:32 PM,10:39 AM,10:19 PM,Waxing Crescent,21
39592,Zambia,Lusaka,-15.4167,28.2833,Africa/Lusaka,1733481900,12/6/2024 12:45,36.0,96.8,Sunny,...,14.245,14.245,1,2,5:28 AM,6:28 PM,9:58 AM,11:06 PM,Waxing Crescent,22


#### Penjelasan:
Kode di atas hanya untuk buka file CSV-nya menggunakan pandas, kemudian akan disimpan ke variabel `df` agar bisa diolah. Baris terakhir (`df`) itu untuk nampilin isinya, sekalian ngecek datanya kebaca dengan benar atau tidak.

In [5]:
korelasi = df[['temperature_fahrenheit', 'pressure_mb', 'air_quality_Sulphur_dioxide']].corr()
korelasi

,temperature_fahrenheit,pressure_mb,air_quality_Sulphur_dioxide
temperature_fahrenheit,1.000000,-0.486833,-0.023152
pressure_mb,-0.486833,1.000000,0.018620
air_quality_Sulphur_dioxide,-0.023152,0.018620,1.000000


#### Penjelasan:
Kode di atas memilih 3 kolom yaitu `temperature_fahrenheit`, `pressure_mb`, dan `air_quality_Sulphur_dioxide` dari DataFrame `df`, kemudian akan dicek korelasinya pakai `.corr()`. Fungsi ini akan memberi tahu seberapa nyambung satu variabel sama variabel lainnya.

Dari hasil di atas terlihat bahwa `temperature_fahrenheit` vs `pressure_mb` bernilai -0.486833 (korelasi negatif sedang, artinya semakin tinggi suhu maka tekanan udara cenderung menurun), sedangkan korelasi `air_quality_Sulphur_dioxide` terhadap dua variabel lainnya sangat lemah (mendekati 0), menandakan hampir tidak ada hubungan.

## Tugas 2 - Bagian 1
Dataset **Customer Shopping Trends**. Analisa `mean`, `std`, `min`, `max`, Q1, Q2, Q3 untuk data `Previous Purchases`.

In [6]:
import pandas as pd
path = "shopping_trends_updated.csv"
df = pd.read_csv(path)
df

,Customer ID,Age,Gender,Item Purchased,Category,Purchase Amount (USD),Location,Size,Color,Season,Review Rating,Subscription Status,Shipping Type,Discount Applied,Promo Code Used,Previous Purchases,Payment Method,Frequency of Purchases
0,1,55,Male,Blouse,Clothing,53,Kentucky,L,Gray,Winter,3.1,Yes,Express,Yes,Yes,14,Venmo,Fortnightly
1,2,19,Male,Sweater,Clothing,64,Maine,L,Maroon,Winter,3.1,Yes,Express,Yes,Yes,2,Cash,Fortnightly
2,3,50,Male,Jeans,Clothing,73,Massachusetts,S,Maroon,Spring,3.1,Yes,Free Shipping,Yes,Yes,23,Credit Card,Weekly
3,4,21,Male,Sandals,Footwear,90,Rhode Island,M,Maroon,Spring,3.5,Yes,Next Day Air,Yes,Yes,49,PayPal,Weekly
4,5,45,Male,Blouse,Clothing,49,Oregon,M,Turquoise,Spring,2.7,Yes,Free Shipping,Yes,Yes,31,PayPal,Annually
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3895,3896,40,Female,Hoodie,Clothing,28,Virginia,L,Turquoise,Summer,4.2,No,2-Day Shipping,No,No,32,Venmo,Weekly
3896,3897,52,Female,Backpack,Accessories,49,Iowa,L,White,Spring,4.5,No,Store Pickup,No,No,41,Bank Transfer,Bi-Weekly
3897,3898,46,Female,Belt,Accessories,33,New Jersey,L,Green,Spring,2.9,No,Standard,No,No,24,Venmo,Quarterly
3898,3899,44,Female,Shoes,Footwear,77,Minnesota,S,Brown,Summer,3.8,No,Express,No,No,24,Venmo,Weekly


#### Penjelasan:
Sama seperti sebelumnya, ini hanya untuk membuka file CSV-nya dan simpan ke `df` biar siap diolah.

In [7]:
statistik_previous_purchases = df['Previous Purchases'].describe()
statistik_previous_purchases

count    3900.000000
mean       25.351538
std        14.447125
min         1.000000
25%        13.000000
50%        25.000000
75%        38.000000
max        50.000000
Name: Previous Purchases, dtype: float64

#### Penjelasan:
Di sini saya pakai `.describe()` untuk melihat ringkasan angka-angka dari kolom `Previous Purchases` sekaligus, jadi tidak perlu hitung satu-satu.

Dari hasilnya, rata-rata pelanggan telah belanja sekitar 25 kali sebelumnya, tapi variasinya lumayan lebar (std-nya 14.45), ada yang baru belanja 1 kali, ada juga yang udah sampai 50 kali. Kalau dilihat dari Q1 sampai Q3, kebanyakan pelanggan (separuhnya) itu telah belanja antara 13 sampai 38 kali.

## Tugas 2 - Bagian 2
Dataset **Customer Shopping Trends**. Analisa korelasi `Age` vs `Purchase Amount (USD)`.

In [8]:
import pandas as pd
path = "shopping_trends_updated.csv"
df = pd.read_csv(path)
df

,Customer ID,Age,Gender,Item Purchased,Category,Purchase Amount (USD),Location,Size,Color,Season,Review Rating,Subscription Status,Shipping Type,Discount Applied,Promo Code Used,Previous Purchases,Payment Method,Frequency of Purchases
0,1,55,Male,Blouse,Clothing,53,Kentucky,L,Gray,Winter,3.1,Yes,Express,Yes,Yes,14,Venmo,Fortnightly
1,2,19,Male,Sweater,Clothing,64,Maine,L,Maroon,Winter,3.1,Yes,Express,Yes,Yes,2,Cash,Fortnightly
2,3,50,Male,Jeans,Clothing,73,Massachusetts,S,Maroon,Spring,3.1,Yes,Free Shipping,Yes,Yes,23,Credit Card,Weekly
3,4,21,Male,Sandals,Footwear,90,Rhode Island,M,Maroon,Spring,3.5,Yes,Next Day Air,Yes,Yes,49,PayPal,Weekly
4,5,45,Male,Blouse,Clothing,49,Oregon,M,Turquoise,Spring,2.7,Yes,Free Shipping,Yes,Yes,31,PayPal,Annually
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3895,3896,40,Female,Hoodie,Clothing,28,Virginia,L,Turquoise,Summer,4.2,No,2-Day Shipping,No,No,32,Venmo,Weekly
3896,3897,52,Female,Backpack,Accessories,49,Iowa,L,White,Spring,4.5,No,Store Pickup,No,No,41,Bank Transfer,Bi-Weekly
3897,3898,46,Female,Belt,Accessories,33,New Jersey,L,Green,Spring,2.9,No,Standard,No,No,24,Venmo,Quarterly
3898,3899,44,Female,Shoes,Footwear,77,Minnesota,S,Brown,Summer,3.8,No,Express,No,No,24,Venmo,Weekly


#### Penjelasan:
Sama juga seperti sebelumnya, untuk membuka file CSV-nya, disimpan ke `df` agar bisa dipakai untuk analisa selanjutnya.

In [9]:
korelasi_age_purchase = df[['Age', 'Purchase Amount (USD)']].corr()
korelasi_age_purchase

,Age,Purchase Amount (USD)
Age,1.000000,-0.010424
Purchase Amount (USD),-0.010424,1.000000


#### Penjelasan:
Kode di atas untuk cek korelasi antara umur pelanggan (`Age`) sama uang yang dia belanjain (`Purchase Amount (USD)`).

Hasilnya cuma -0.01, itu sudah sangat dekat ke 0. Jadi bisa dibilang umur tidak ngaruh sama sekali ke besar-kecilnya belanja. Mau muda atau tua, nominal belanjanya ya kurang lebih sama aja.

## Tugas 2 - Bagian 3
Dataset **Customer Shopping Trends**. Analisa korelasi `Purchase Amount (USD)` vs `Review Rating`.

In [10]:
import pandas as pd
path = "shopping_trends_updated.csv"
df = pd.read_csv(path)
df

,Customer ID,Age,Gender,Item Purchased,Category,Purchase Amount (USD),Location,Size,Color,Season,Review Rating,Subscription Status,Shipping Type,Discount Applied,Promo Code Used,Previous Purchases,Payment Method,Frequency of Purchases
0,1,55,Male,Blouse,Clothing,53,Kentucky,L,Gray,Winter,3.1,Yes,Express,Yes,Yes,14,Venmo,Fortnightly
1,2,19,Male,Sweater,Clothing,64,Maine,L,Maroon,Winter,3.1,Yes,Express,Yes,Yes,2,Cash,Fortnightly
2,3,50,Male,Jeans,Clothing,73,Massachusetts,S,Maroon,Spring,3.1,Yes,Free Shipping,Yes,Yes,23,Credit Card,Weekly
3,4,21,Male,Sandals,Footwear,90,Rhode Island,M,Maroon,Spring,3.5,Yes,Next Day Air,Yes,Yes,49,PayPal,Weekly
4,5,45,Male,Blouse,Clothing,49,Oregon,M,Turquoise,Spring,2.7,Yes,Free Shipping,Yes,Yes,31,PayPal,Annually
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3895,3896,40,Female,Hoodie,Clothing,28,Virginia,L,Turquoise,Summer,4.2,No,2-Day Shipping,No,No,32,Venmo,Weekly
3896,3897,52,Female,Backpack,Accessories,49,Iowa,L,White,Spring,4.5,No,Store Pickup,No,No,41,Bank Transfer,Bi-Weekly
3897,3898,46,Female,Belt,Accessories,33,New Jersey,L,Green,Spring,2.9,No,Standard,No,No,24,Venmo,Quarterly
3898,3899,44,Female,Shoes,Footwear,77,Minnesota,S,Brown,Summer,3.8,No,Express,No,No,24,Venmo,Weekly


#### Penjelasan:
Sama seperti sebelumnya, kode di atas untuk membuka lagi datanya, disimpan ke `df`, agar bisa dipakai untuk cek korelasi berikutnya.

In [11]:
korelasi_purchase_rating = df[['Purchase Amount (USD)', 'Review Rating']].corr()
korelasi_purchase_rating

,Purchase Amount (USD),Review Rating
Purchase Amount (USD),1.000000,0.030776
Review Rating,0.030776,1.000000


#### Penjelasan:
Yang terakhir ini, untuk mencek apakah nominal belanja (`Purchase Amount (USD)`) ada hubungannya sama rating yang dikasih pelanggan (`Review Rating`).

Hasilnya juga kecil, cuma 0.03, jadi nyaris nggak ada hubungan. Artinya orang yang belanja banyak belum tentu ngasih rating tinggi, dan sebaliknya, orang yang belanja dikit juga bisa aja ngasih rating bagus.